bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValueImpl.java

```diff
@@ -840,15 +848,15 @@ protected void rangeCheck(long index, int size) {
 
     @Override
     protected void fillerValueCheck(int index, int size) {
-        // if there has been values added beyond the current index, that means filler values
-        // has already been checked. Therefore no need to check again.
-        if (this.size >= index) {
-            return;
+        if (!hasFillerValue.isPresent()) {
```

Comment:
This `fillerValueCheck()` method is called for each and every add operation. So could this be an overhead?

Since `hasFillerValue` needs to be calculated only once for a given type, can we move this flag to the array-type, and have it calculated only once during the initialisation of the type?

Reply:
fixed by https://github.com/ballerina-platform/ballerina-lang/pull/20749/commits/c0998bf94dc83e767795d8135dc411e1605c6405